src/app/shared/services/piece-layer.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Piece } from '../../core/models/piece'
+import {Part} from "../../core/models/part";
+
+export class PieceLayerService {
+    public getPieceLayersForParts(parts: Part[]): string[] {
+        const pieceLayers: string[] = []
+        parts.forEach(part => {
+            part.pieces.concat(part.adLibPieces).forEach(piece => {
+                const layer = this.getPieceLayer(piece)
+                if (!pieceLayers.includes(layer) && layer !== 'UNKNOWN') {
+                    pieceLayers.push(layer)
+                }
+            })
+        })
+
+        return pieceLayers
+    }
+
+    //TODO: Does not account for the fact that a JINGLE should be PGM when in the KLAR ON AIR (if this is desired behaviour?)
+    public getPieceLayer(piece: Piece): string {
```

Comment:
I can already tell that ones we add the `complexity` linting rule, with a max of 10, that this method is going to error out.
Is there some way it could be split down into smaller methods, to reduce the cyclomatic complexity?

Reply:
I can tell that the cyclomatic complexity of this method is 7, so it is within the desired threshold.
And I can contribute with the knowledge that this entire method will be rewritten completely when `piece.type` is being populated in the backend.